pyomo/contrib/doe/fim_doe_tutorial.ipynb

```diff
@@ -0,0 +1,2994 @@
+      "Reallocating memory for MA57: lfact (333406)\n",
+      "   4  6.8268725e+00 9.79e+04 8.00e+09  -1.0 1.26e+00    -  4.97e-01 4.76e-01h  1\n",
+      "   5  6.8275075e+00 6.50e+04 6.37e+09  -1.0 9.61e-01   9.6 1.00e+00 3.70e-01h  1\n",
+      "Reallocating memory for MA57: lfact (350926)\n",
+      "   6  6.8272633e+00 5.99e+04 5.94e+09  -1.0 7.83e-01   9.1 1.00e+00 8.01e-02h  1\n",
+      "   7  6.8215341e+00 5.11e+04 5.11e+09  -1.0 7.51e-01   8.6 1.00e+00 1.53e-01h  1\n",
+      "   8  6.7925573e+00 4.39e+04 4.40e+09  -1.0 6.94e-01   8.1 1.00e+00 1.46e-01h  1\n",
+      "   9  6.6660412e+00 3.65e+04 3.66e+09  -1.0 6.43e-01   7.7 1.00e+00 1.77e-01h  1\n",
+      "iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls\n",
+      "  10  6.0539190e+00 2.69e+04 2.71e+09  -1.0 5.86e-01   7.2 1.00e+00 2.81e-01h  1\n",
+      "  11  5.7667182e+00 1.32e+04 1.34e+09  -1.0 5.04e-01   7.6 1.00e+00 5.99e-01h  1\n",
+      "  12  4.8051241e+00 3.30e+03 3.36e+08  -1.0 4.35e-01   7.1 2.83e-01 1.00e+00h  1\n",
+      "  13 -1.9483298e+01 4.06e+04 1.15e+11  -1.0 1.29e+04    -  2.98e-05 1.56e-02f  7\n",
+      "Reallocating memory for MA57: lfact (390493)\n",
+      "  14 -1.6564376e+01 1.02e+04 7.34e+12  -1.0 1.00e+02  11.2 1.00e+00 1.00e+00h  1\n",
+      "  15 -1.4422590e+01 2.72e+03 1.17e+13  -1.0 4.81e+01  11.6 1.00e+00 1.00e+00h  1\n",
+      "  16 -1.1183342e+01 5.77e+02 4.66e+12  -1.0 2.66e+01  11.1 1.00e+00 1.00e+00h  1\n",
+      "  17 -9.4794415e+00 1.41e+02 3.24e+12  -1.0 9.83e+00  11.5 1.00e+00 1.00e+00h  1\n",
+      "  18 -6.5391561e+00 7.14e+01 1.30e+12  -1.0 7.94e+00  11.1 1.00e+00 1.00e+00h  1\n",
+      "  19 -5.6255802e+00 9.49e+01 5.10e+11  -1.0 2.06e+00  11.5 1.00e+00 1.00e+00H  1\n",
+      "iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls\n",
+      "  20 -5.2916246e+00 9.90e+01 4.11e+11  -1.0 1.05e+00  11.0 9.41e-01 2.50e-01h  3\n",
+      "  21 -5.1397237e+00 1.00e+02 3.32e+11  -1.0 3.88e-01  11.4 1.00e+00 2.50e-01h  3\n",
+      "  22 -4.3201634e+00 7.67e+01 2.27e+11  -1.0 9.35e-01  11.0 7.66e-01 1.00e+00H  1\n",
+      "  23 -4.1623141e+00 4.27e+01 4.43e+11  -1.0 1.72e+00  11.4 1.00e+00 1.00e+00h  1\n",
+      "  24 -3.6214324e+00 7.89e+01 2.63e+11  -1.0 4.64e+00  10.9 1.00e+00 1.00e+00h  1\n",
+      "  25 -2.9410001e+00 6.03e+01 1.18e+11  -1.0 6.17e-01  11.3 1.00e+00 1.00e+00h  1\n",
+      "  26 -2.8503950e+00 5.63e+01 2.42e+11  -1.0 2.28e+01  10.9 3.98e-02 6.96e-02H  1\n",
+      "  27 -2.7476817e+00 5.12e+01 2.54e+11  -1.0 1.75e+00  11.3 1.00e+00 1.37e-01h  2\n",
+      "  28r-2.7476817e+00 5.12e+01 1.00e+03   1.7 0.00e+00  11.7 0.00e+00 3.60e-07R 21\n",
+      "  29r-3.5240709e+00 2.90e+01 9.98e+02   1.7 5.33e+04    -  2.46e-03 9.51e-04f  1\n",
+      "iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls\n",
+      "  30r-4.2317368e+00 2.28e+01 9.94e+02   1.0 6.31e+03    -  4.58e-03 3.00e-03f  1\n",
```

Comment:
Suggestion: stop Ipopt at iteration 30 and check which constraints are infeasible. That may be a hint as to why the problem is stuck in restoration.